pkg/k8shandler/collection.go

```diff
@@ -0,0 +1,166 @@
+package k8shandler
+
+import (
+  "k8s.io/apimachinery/pkg/api/errors"
+  "github.com/sirupsen/logrus"
+  "github.com/openshift/cluster-logging-operator/pkg/utils"
+  "k8s.io/api/core/v1"
+
+  sdk "github.com/operator-framework/operator-sdk/pkg/sdk"
+  logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1alpha1"
+)
+
+func CreateOrUpdateCollection(logging *logging.ClusterLogging) error {
+  createOrUpdateFluentdServiceAccount(logging)
+
+  createOrUpdateCollectionPriorityClass(logging)
+  createOrUpdateFluentdConfigMap(logging)
+  createOrUpdateFluentdDaemonset(logging)
+
+  return createOrUpdateFluentdSecret(logging)
+}
+
+func createOrUpdateCollectionPriorityClass(logging *logging.ClusterLogging) error {
+
+  collectionPriorityClass := utils.PriorityClass("cluster-logging", 1000000, false, "This priority class is for the Cluster-Logging Collector")
+
+  sdk.Create(collectionPriorityClass)
+  // Currently we cannot do validation checking on this due to version issues with the scheduling api and the version we create the priorityclass with
+  /*err := sdk.Create(collectionPriorityClass)
+  if err != nil && !errors.IsAlreadyExists(err) {
+    logrus.Fatalf("Failure creating Collection priority class: %v", err)
+  }*/
+
+  return nil
+}
+
+func createOrUpdateFluentdServiceAccount(logging *logging.ClusterLogging) error {
+
+  fluentdServiceAccount := utils.ServiceAccount("aggregated-logging-fluentd", logging.Namespace)
```

Comment:
collector in lieu of aggregated-logging-fluentd  ?

Reply:
per the discussion with @richm on IRC, we would want to be explicit about the service account name, as it is more straight forward and easier when it comes to docs/support